src/main/java/edu/kit/compiler/optimizations/ConstantOptimization.java

```diff
@@ -126,28 +133,50 @@ private boolean transformMemoryProj(Proj node) {
      */
     private boolean transformControlFlowProj(Proj node) {
         Node pred = node.getPred();
-        TargetValueLatticeElement predValue;
-        if (pred instanceof Cond && (predValue = nodeValues.get(pred)).isConstant()) {
-            if (
-                (node.getNum() == pn_Cond.pn_Cond_false.val && predValue.getValue().equals(TargetValue.getBFalse())) ||
-                (node.getNum() == pn_Cond.pn_Cond_true.val && predValue.getValue().equals(TargetValue.getBTrue()))
-            ) {
-                Node jmpNode = graph.newJmp(node.getBlock());
-                Graph.exchange(node, jmpNode);
-
-                return true;
-            } else if (
-                (node.getNum() == pn_Cond.pn_Cond_false.val && predValue.getValue().equals(TargetValue.getBTrue())) ||
-                (node.getNum() == pn_Cond.pn_Cond_true.val && predValue.getValue().equals(TargetValue.getBFalse()))
-            ) {
-                Node badNode = graph.newBad(Mode.getX());
-                Graph.exchange(node, badNode);
-
-                return true;
+        if (pred instanceof Cond) {
+            TargetValueLatticeElement predValue = nodeValues.getOrDefault(pred, TargetValueLatticeElement.unknown());
+
+            if (predValue.isUnknown()) {
+                predValue = TargetValueLatticeElement.constant(undefinedCondStrategy.chooseCondValue((Cond)pred));
+                nodeValues.put(pred, predValue);
+            }
+
+            if (predValue.isConstant()) {
+                if (
+                    (node.getNum() == pn_Cond.pn_Cond_false.val && predValue.getValue().equals(TargetValue.getBFalse())) ||
+                    (node.getNum() == pn_Cond.pn_Cond_true.val && predValue.getValue().equals(TargetValue.getBTrue()))
```

Comment:
Is there a reason you're not using `Cond.pn{True,False}`?

Reply:
No, I just copied it from where I used this before :D